### DISCUSSION TITLE
Wake cron dont work

### DISCUSSION THREAD

**Discussion by pedropetz** [seriousm4x/UpSnap#506] [Help wanted] [answered]
Hi everyone,

I've set up a device and I can successfully wake up the server in question using the button on dashboard.
However, I configure the "Wake cron" option at the time interval "* 11,19 * * *" and it doesn't wake up the server.
I didn't configure the "Custom wake command" because I didn't think it was necessary.

Can anyone help me figure out if I'm missing a configuration or help me diagnose the problem?

**Comment by kraizelburg** [comment]
Hi, I can confirm wake cron does not work for me either in the latest version. I have it set up to 00 09 * * * and does not wake up the server. I didn't configure the custom wake either.

**Comment by seriousm4x** [comment]
Im currently on vacation but will look into it once im back.

**Reply by kraizelburg** [reply]
Today I found something else that may be a clue about what is happening. 
My current host is rpi4 with upsnap installed with docker and I use it for waking up my debian 12 server. Yesterday, I triggered the wake up manually via upsnap UI and I saved wake cron 01 08 * * * today, the server didn't wake up  and docker logs had no trace of the wake request, then I quickly saved another cron 15 08 * * * and it woke up the server! this time docker showed the wake request correctly. 
It seems that after 1 day if you manually save a new wake cron in the UI it triggers it but not the next day. 
Anyway thank you

**Comment by pedropetz** [comment]
I was using a LXC from Proxmox and used the "UpSnap_4.2.8_linux_amd64" had this problem. Today I configure on a raspberry pi natively with "UpSnap_4.2.8_linux_armv7" and It works. No problem

**Comment by kraizelburg** [comment]
I just downgraded to version 4.2.1 and now it seems to work just fine. So I guess there is something odd with the docker image in a release after 4.2.1.

Edit: false alarm, today cron work didn't trigger again :(

**Comment by kraizelburg** [comment]
Another update: after the failing downgrade to version 4.2.1 I upgraded to the latest label and cron isn't working yet but I saw in docker logs that crons are not trigerred at all except when you manually change the cron and save but next day it will fail.

**Comment by pedropetz** [comment]
Another update: Stop working again... I will downgrade and check which version cron works.
Is not Docker related, a native binary have the same behavier

**Reply by kraizelburg** [reply]
Yes, this morning cron wasn´t triggered for  me either. The version that was working fine was upsnap version 3. 
If you find a version that works please post it. Thanks

**Comment by seriousm4x** [comment]
I'm back from vacation. Strangely enough, cron has been working for me for the last 3 days since setting it up. I' using a wake cron job for one of my devices (`0 12 * * *`). Running in docker on my unraid system.

However, I have found a possible solution to this. When some device values were changed, the cron scheduler was not restarted correctly.

Maybe [4.2.9-beta.1](https://github.com/seriousm4x/UpSnap/releases/tag/4.2.9-beta.1) fixes this. Please try.

**Reply by kraizelburg** [reply]
I pulled the beta image you mentioned and I will update in the coming days. but anyway your solution makes total sense.

**Reply by kraizelburg** [reply]
Hi, today cron was not triggered again so I guess there is still something odd. I will revert to upsnap version 3

**Reply by seriousm4x** [reply]
So I tried reproducing this for a week now and I'm not able to. Would someone with this problem mind to share their pb_data folder with me to debug this?

**Comment by seriousm4x** [comment, marked as answer]
I think I found the issue. At least I could reproduce it. The problem was, that the cron worker had an old status of the device, therefore not triggering the wake/shutdown correctly.
Everyone please try [4.2.9-beta.2](https://github.com/seriousm4x/UpSnap/releases/tag/4.2.9-beta.2) @pedropetz @kraizelburg

**Reply by pedropetz** [reply]
I will try it

**Reply by pedropetz** [reply]
with 0 11,19 * * * woke up at 11:00 and 19:00
I think it's working now. thanks

**Reply by seriousm4x** [reply]
Nice, I've pushed a new release

**Reply by kraizelburg** [reply]
Hi, thanks a lot for fixing the problem. I have tested it today and it seems to work as expected.

**Reply by kraizelburg** [reply]
Another update, today wake up cron did not trigger.  What logs can I send to fix the issue? btw I am using docker in rpi4 to wak a local pc

**Reply by seriousm4x** [reply]
Ah really? Crap.. I’ll have another look at this. This wouldn’t be logged I think. So nothing to send me.

**Reply by fun4stuff** [reply]
Cron doesn't seem to be working on 4.6.0

**Reply by seriousm4x** [reply]
> Cron doesn't seem to be working on 4.6.0

This discussion is not related to your problem. See https://github.com/seriousm4x/UpSnap/issues/975#issuecomment-2641527988
